### PR TITLE
jabref: Add version 5.1

### LINKED ISSUE #3164: [Request] Add JabRef
JabRef: A graphical Java application for managing BibTeX and biblatex (.bib) databases.

https://github.com/JabRef/jabref

### PR DESCRIPTION
**Pull request by adrianocorrea**
- Closes #3164

### CODE REVIEW

File: bucket/jabref.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{
+    "version": "5.1",
+    "description": "Free Reference Manager",
```
Comment (Ash258) [inline bucket/jabref.json:3]: Completely saying-nothing description.

What references?? File? Dependencies? Software?

File: bucket/jabref.json [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    "homepage": "https://www.jabref.org",
+    "license": {
+        "identifier": "MIT",
+        "url": "https://github.com/JabRef/jabref/blob/master/LICENSE.md"
+    },
```
Comment (Ash258) [inline bucket/jabref.json:8]: license: MIT

File: bucket/jabref.json [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    "bin": "JabRef.exe",
+    "shortcuts": [
+        [
+            "JabRef.exe",
+            "JabRef"
+        ]
+    ],
+    "checkver": {
+        "url": "https://www.fosshub.com/JabRef.html",
+        "regex": "JabRef-([\\d.]+)\\-portable_windows.zip"
```
Comment (Ash258) [inline bucket/jabref.json:21]: `\\.zip`

File: bucket/jabref.json [hunk 4]
```diff
@@ -0,0 +1,26 @@
+{
+    "version": "5.1",
+    "description": "Free Reference Manager",
+    "homepage": "https://www.jabref.org",
+    "license": {
+        "identifier": "MIT",
+        "url": "https://github.com/JabRef/jabref/blob/master/LICENSE.md"
+    },
+    "url": "https://www.fosshub.com/JabRef.html/JabRef-5.1-portable_windows.zip",
+    "hash": "1a0aaa60c7424a356e3e288cc27218ddd5fe8e85ae295d87a561960404318666",
```
Comment (Ash258) [inline bucket/jabref.json:10]: There is not a single reason to prefer fosshub over github

**Review by Ash258 — CHANGES_REQUESTED** [review]

File: bucket/jabref.json (hunk 1 above)
Comment (adrianocorrea) [reply bucket/jabref.json]: Updated in new commit.

Comment (adrianocorrea) [reply bucket/jabref.json]: Updated. See commit "Update jabref.json"

File: bucket/jabref.json (hunk 2 above)
Comment (adrianocorrea) [reply bucket/jabref.json]: Updated. See commit "Update jabref.json"

File: bucket/jabref.json (hunk 3 above)
Comment (adrianocorrea) [reply bucket/jabref.json]: No longer relevant; replaced fosshub with github.

File: bucket/jabref.json (hunk 4 above)
Comment (adrianocorrea) [reply bucket/jabref.json]: Updated. See commit "Update jabref.json"

**Review by Ash258 — CHANGES_REQUESTED** [review]
Where is autoupdate?

File: bucket/jabref.json [hunk 5]
```diff
@@ -0,0 +1,22 @@
+{
+    "version": "5.1",
+    "description": "Open-sourced, cross-platform citation and reference management software that uses BibTeX and BibLaTeX as its native formats and is therefore typically used for LaTeX",
```
Comment (Ash258) [inline bucket/jabref.json:3]:
```suggestion
    "description": "Citation and reference management software that uses BibTeX and BibLaTeX as its native formats",
```

File: bucket/jabref.json [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+    "homepage": "https://www.jabref.org",
+    "license": "MIT",
+    "url": "https://github.com/JabRef/jabref/releases/download/v5.1/JabRef-5.1-portable_windows.zip",
+    "hash": "1a0aaa60c7424a356e3e288cc27218ddd5fe8e85ae295d87a561960404318666",
```
Comment (Ash258) [inline bucket/jabref.json:7]:
```suggestion
    "architecture": {
        "64bit": {
            "url": "https://github.com/JabRef/jabref/releases/download/v5.1/JabRef-5.1-portable_windows.zip",
            "hash": "1a0aaa60c7424a356e3e288cc27218ddd5fe8e85ae295d87a561960404318666"
        }
    },
```

File: bucket/jabref.json [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+        [
+            "JabRef.exe",
+            "JabRef"
+        ]
+    ],
+    "checkver": {
+        "github": "https://github.com/JabRef/jabref"
+    },
+    "autoupdate": {
+        "url": "https://github.com/JabRef/jabref/releases/download/v$version/JabRef-$version-portable_windows.zip"
```
Comment (Ash258) [inline bucket/jabref.json:20]:
```suggestion
        "architecture": {
            "64bit": {
                "url": "https://github.com/JabRef/jabref/releases/download/v$version/JabRef-$version-portable_windows.zip"
            }
        }
```

File: bucket/jabref.json [hunk 8]
```diff
@@ -0,0 +1,22 @@
+{
+    "version": "5.1",
+    "description": "Open-sourced, cross-platform citation and reference management software that uses BibTeX and BibLaTeX as its native formats and is therefore typically used for LaTeX",
+    "homepage": "https://www.jabref.org",
+    "license": "MIT",
+    "url": "https://github.com/JabRef/jabref/releases/download/v5.1/JabRef-5.1-portable_windows.zip",
+    "hash": "1a0aaa60c7424a356e3e288cc27218ddd5fe8e85ae295d87a561960404318666",
+    "extract_dir": "JabRef",
+    "bin": "JabRef.exe",
```
Comment (Ash258) [inline bucket/jabref.json:9]: What are available parameters?

Any of [these](https://github.com/JabRef/jabref/blob/master/src/main/java/org/jabref/cli/JabRefCLI.java) do not work

**Review by Ash258 — CHANGES_REQUESTED** [review]

File: bucket/jabref.json (hunk 8 above)
Comment (adrianocorrea) [reply bucket/jabref.json:13]: Available parameters are: 

- Help: -h
- No-GUI mode: -n
- Import file: -i filename[,import format]
- Export file: -o filename[,export format]
- Import BibTeX: -importBibtex
- Export matching entries: -m [field]searchTerm,outputFile:file[,exportFormat]
- Fetch entries from Web: -f=FetcherName:QueryString
- Subdatabase from .aux file: -a infile[.aux],outfile[.bib] base-BibTeX-file
- Set file links: -asfl
- Regenerate keys: -g
- Export preferences: -x filename
- Import preferences: -p filename
- Reset preferences: -d key
- No files at startup: -b
- Version: -v
- Debug mode: --debug
- Display output in the console: --console

You need to use JabRef.bat for CLI (see the project's [documentation](https://docs.jabref.org/advanced/commandline).

Comment (Ash258) [reply bucket/jabref.json:13]: Then it is nonsense to expose jabref.exe instead of the bat...

Comment (adrianocorrea) [reply bucket/jabref.json]: Ok. JabRef.exe was replaced with JabRef.bat. You might see a message "ERROR StatusLogger Unrecognized format specifier" in the terminal, which is a known issue (see issue [#6009](https://github.com/JabRef/jabref/issues/6009) with JabRef CLI.

**Review by Ash258 — APPROVED** [review]

**Pull request merged**